Translate architecture concepts in French (#12882)
kubernetes/website: content/fr/docs/concepts/architecture/master-node-communication.md

---
reviewers:
- sieben

> title: Communication Master-Node

Reviewer: ```suggestion
title: Communication Master-Nœud
```
ou _Communication Maître-Nœud_?

Author: Je serais d'avis de rester avec une cohérence de langue là dessus

content_template: templates/concept
weight: 20
---

{{% capture overview %}}

> Ce document répertorie les cannaux de communication entre l'API du noeud maître (apiserver of master node en anglais) et le rester du cluster Kubernetes.

Reviewer 2: Typo
```suggestion
Ce document répertorie les canaux de communication entre l'API du noeud maître (apiserver of master node en anglais) et le rester du cluster Kubernetes.
```

{{% /capture %}}

{{% capture body %}}

## Communication du Cluster vers Master

Tous les canaux de communication du cluster au master se terminent au apiserver (aucun des autres composants principaux n'est conçu pour exposer des services distants).
Dans un déploiement typique, l'apiserver est configuré pour écouter les connexions distantes sur un port HTTPS sécurisé (443) avec un ou plusieurs types d'[authentification](/docs/reference/access-authn-authz/authentication/) client.
Une ou plusieurs formes d'[autorisation](/docs/reference/access-authn-authz/authorization/) devraient être activée, notamment si les [requêtes anonymes](/docs/reference/access-authn-authz/authentication/#anonymous-requests) ou [jeton de compte de service](/docs/reference/access-authn-authz/authentication/#service-account-tokens) sont autorisés.

Le certificat racine public du cluster doit être configuré pour que les nœuds puissent se connecter en toute sécurité à l'apiserver avec des informations d'identification client valides.
Par exemple, dans un déploiement GKE par défaut, les informations d'identification client fournies au kubelet sont sous la forme d'un certificat client.

> Consultez [kubelet TLS bootstrapping](/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/) pour le provisioning automatisé des certificats de client Kubelet.

Reviewer: ```suggestion
Consultez [amorçage TLS de kubelet](/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/) pour le provisioning automatisé des certificats de client Kubelet.
```

Les pods qui souhaitent se connecter à l'apiserver peuvent le faire de manière sécurisée en utilisant un compte de service afin que Kubernetes injecte automatiquement le certificat racine public et un jeton de support valide dans le pod lorsqu'il est instancié.
Le service `kubernetes` (dans tous les namespaces) est configuré avec une adresse IP virtuelle redirigée (via kube-proxy) vers le point de terminaison HTTPS sur le apiserver.

Les composants du master communiquent également avec l'apiserver du cluster via le port sécurisé.

Par conséquent, le mode de fonctionnement par défaut pour les connexions du cluster (nœuds et pods s'exécutant sur les nœuds) au master est sécurisé par défaut et peut s'exécuter sur des réseaux non sécurisés et/ou publics.

> ## Communication du Master vers Cluster

Reviewer 3: vers Cluster => vers le Cluster ?

Reviewer: Master → Maître?

> Il existe deux voies de communication principales du master (apiserver) au cluster.

Reviewer: ```suggestion
Il existe deux voies de communication principales du maître (apiserver) au cluster.
```

> Le premier est du processus apiserver au processus kubelet qui s'exécute sur chaque nœud du cluster.

Reviewer 4: "la premiere"

La seconde part de l'apiserver vers n'importe quel nœud, pod ou service via la fonctionnalité proxy de l'apiserver.

### Communication de l'apiserver vers le kubelet

Les connexions de l'apiserver au kubelet sont utilisées pour:

  * Récupérer les logs des pods.
  * S'attacher (via kubectl) à des pods en cours d'exécution.
  * Fournir la fonctionnalité de transfert de port du kubelet.

Ces connexions se terminent au point de terminaison HTTPS du kubelet.

> Par défaut, l'apiserver ne vérifie pas le certificat du kubelet, ce qui rend la connexion sujette aux attaques de type "man-in-the-middle", et **non sûre** sur des réseaux non approuvés et/ou publics.

Reviewer: ```suggestion
Par défaut, l'apiserver ne vérifie pas le certificat du kubelet, ce qui rend la connexion sujette aux attaques de l'homme du milieu (man-in-the-middle attack en anglais), et **non sûre** sur des réseaux non approuvés et/ou publics.
```

Author: Personne n'utilise homme du milieu dans ce contexte.

Pour vérifier cette connexion, utilisez l'argument `--kubelet-certificate-authority` pour fournir à apiserver un ensemble de certificats racine à utiliser pour vérifier le certificat du kubelet.

> Si ce n'est pas possible, utilisez [SSH tunneling](/docs/tasks/access-application-cluster/port-forward-access-application-cluster/) entre l'apiserver et le kubelet si nécessaire pour éviter la connexion sur un réseau non sécurisé ou public.

Reviewer: ```suggestion
Si ce n'est pas possible, utilisez [tunneling SSH](/docs/tasks/access-application-cluster/port-forward-access-application-cluster/) entre l'apiserver et le kubelet si nécessaire pour éviter la connexion sur un réseau non sécurisé ou public.
```
ou _tunnelage SSH_

> Finalement, l'[authentification et/ou autorisation  du Kubelet](/docs/admin/kubelet-authentication-authorization/) devrait être activé pour sécuriser l'API kubelet.

Reviewer 2: Double espace 
```suggestion
Finalement, l'[authentification et/ou autorisation du Kubelet](/docs/admin/kubelet-authentication-authorization/) devrait être activé pour sécuriser l'API kubelet.
```
